lib/api/controllers/securityController.js

```diff
@@ -446,19 +475,7 @@ class SecurityController {
    * @returns {Promise<Object>}
    */
   createRestrictedUser(request) {
-    assertHasBody(request);
-    assertBodyHasNotAttribute(request, 'profileIds');
-    assertIdStartsNotUnderscore(request);
-
-    const pojoUser = request.input.body;
-    pojoUser._id = request.input.resource._id || uuid.v4();
-    pojoUser.profileIds = this.kuzzle.config.security.restrictedProfileIds;
-
-    const user = new User();
-
-    return this.kuzzle.repositories.user.hydrate(user, pojoUser)
-      .then(modifiedUser => this.kuzzle.repositories.user.persist(modifiedUser, {database: {method: 'create'}}))
-      .then(modifiedUser => formatProcessing.formatUserForSerialization(this.kuzzle, modifiedUser));
+    return createUser(this.kuzzle, request, true);
```

Comment:
nipticling: could just inject the default profileIds to the request and execute `this.createUser` (see also comment above):
```javascript
createRestrictedUser(request) {
  assertHasBody(request);
  assertIdStartsNotUnderscore(request);
  assertBodyHasAttribute(request, 'content');
  assertBodyAttributeType(request, 'content', 'object');

  // new assertions method to create:
  assertContentHasNotAttribute(request.input.body.content, 'profileIds');

  request.input.body.content.profileIds = this.kuzzle.config.security.restrictedProfileIds;
  return this.createUser(request);
}
```